### PR TITLE
doc: record/replay python package versions

### PR DESCRIPTION
**Pull request by matthiasdiener**
Depends on https://github.com/illinois-ceesd/emirge/pull/83

### CODE REVIEW

File: doc/faq.rst [hunk 1]
```diff
@@ -87,3 +87,20 @@ You can build pyopencl against conda's OpenCL driver in the following way::
...
+exact versions of the packages at a later time::
+
+   $ cd emirge/
+   $ ./version.sh
+   [...]
+   *** Creating requirements file with current emirge module versions
+   [...]
+   *** Created file 'requirements.txt.XXXXXX'. Install it with 'pip install --src . -r requirements.txt.XXXXXX'.
+
+   $ pip install --src . -r requirements.txt.XXXXXX
```
Comment (inducer) [inline doc/faq.rst:106]: See https://github.com/illinois-ceesd/emirge/pull/83. I think "fresh env" will be safer, more reproducible, and more extensible to when the procedure includes conda packages.

**Review by inducer — COMMENTED** [review]
Thanks. As I said in https://github.com/illinois-ceesd/emirge/pull/83#pullrequestreview-521777909, we should make it so that the command line procedure doesn't change too drastically when we start incorporating conda env files.

File: doc/faq.rst [hunk 2]
```diff
@@ -87,3 +89,27 @@ You can build pyopencl against conda's OpenCL driver in the following way::
...
+exact versions of the packages at a later time::
+
+   $ cd emirge/
+   $ ./version.sh --output-requirements=myreq.txt
+   [...]
+   *** Creating requirements file with current emirge module versions
+   [...]
+   *** Created file 'myreq.txt'. Install it with 'pip install --src . -r myreq.txt'.
+
+   $ pip install --src . -r myreq.txt
```
Comment (inducer) [inline doc/faq.rst:108]: Also mention that the same `requirements.txt` can be used to create a new environment. Looks good otherwise.

Comment (matthiasdiener) [reply doc/faq.rst:108]: added.

File: doc/faq.rst [hunk 3]
```diff
@@ -87,3 +89,35 @@ You can build pyopencl against conda's OpenCL driver in the following way::
...
+   see :ref:`Pyopencl source installation` for information on the prerequisites of Pyopencl,
+   or remove/comment the pyopencl line from the generated requirements file.
+
+.. note::
+
+   You can also install the packages from the created requirements.txt into a new conda environment::
+   
+   $ conda create -n new_env
+   $ conda activate new_env
+   $ pip install --src . -r myreq.txt
```
Comment (inducer) [inline doc/faq.rst:123]: This block needs indenting in order to render properly.

Comment (inducer) [inline doc/faq.rst:123]: I figured that the `requirements.txt` could be provided to `install.sh` via a flag?

Comment (matthiasdiener) [reply doc/faq.rst]: Fixed.

Comment (matthiasdiener) [reply doc/faq.rst]: No, that won't work for now. Our parser doesn't handle the `--editable` flag inside the requirements.txt.

**Comment by matthiasdiener** [comment]
Rendered html is here: https://mirgecom.readthedocs.io/en/doc-req-versions.sh/faq.html

File: doc/faq.rst (hunk 3 above)
Comment (inducer) [reply doc/faq.rst]: It should. Otherwise why are we building systems that don't make sense and where nothing fits together?

Comment (matthiasdiener) [reply doc/faq.rst]: Done in https://github.com/illinois-ceesd/emirge/pull/88

**Review by inducer — APPROVED** [review]
Thanks!

**Pull request merged**